### PR TITLE
Feat/indicator autofit

### PR DESCRIPTION
**Pull request by purpose233**
[[中文版模板 / Chinese template](https://github.com/VisActor/vrender/blob/main/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Release
- [ ] Other (about what?)

### 💡 Background and solution

Support fit strategy for indicator.

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

**Review by xile611 — COMMENTED** [review]
补充一下 vrender 的 bugserver case吧

File: packages/vrender-components/src/indicator/indicator.ts [hunk 1]
```diff
@@ -204,4 +205,83 @@ export class Indicator extends AbstractComponent<Required<IndicatorAttributes>>
...
+    const hx = (-b + Math.sqrt(b ** 2 - 4 * a * c)) / (2 * a);
+    const hy = ra * hx + h;
+    const y = 2 * hy;
+    const lineHeight = (y - otherHeight) / autoFitTexts.length;
+    if (isValidNumber(y)) {
+      autoFitTexts.forEach(textItem => {
+        textItem.text.setAttribute('fontSize', lineHeight);
+        textItem.text.setAttribute(
+          'lineHeight',
+          isValid(textItem.spec.style?.lineHeight) ? textItem.spec.style?.lineHeight : lineHeight
```
[the remark this answers is not in the document]
Comment (purpose233) [reply packages/vrender-components/src/indicator/indicator.ts]: Fixed.

File: packages/vrender-components/src/indicator/indicator.ts [hunk 2]
```diff
@@ -204,4 +205,83 @@ export class Indicator extends AbstractComponent<Required<IndicatorAttributes>>
...
+
+    const r = limit / 2;
+
+    // unify the initial font size for auto fit texts
+    const singleHeight = 12;
+    const autoFitTexts: { text: IText; spec: IndicatorItemSpec }[] = [];
+    // other text height
+    let otherHeight = 0;
+    // non auto fit title height
+    if (this.attribute.title?.autoFit && this.attribute.title?.fitStrategy === 'inscribed') {
```
[the remark this answers is not in the document]
Comment (purpose233) [reply packages/vrender-components/src/indicator/indicator.ts]: Fixed.

File: packages/vrender-components/src/indicator/indicator.ts [hunk 3]
```diff
@@ -204,4 +205,83 @@ export class Indicator extends AbstractComponent<Required<IndicatorAttributes>>
...
+    // max width for all auto fit texts
+    const maxWidth = autoFitTexts.reduce((width, textItem) => {
+      return Math.max(width, textItem.text.AABBBounds.width());
+    }, 0);
+
+    // y = x * (singleHeight / maxWidth * textCount) + otherHeight
+    // hy = hx * (singleHeight / maxWidth * textCount) + otherHeight / 2
+    const ra = (singleHeight / maxWidth) * autoFitTexts.length;
+    const h = otherHeight / 2;
+    const a = ra ** 2 + 1;
```
[the remark this answers is not in the document]
Comment (purpose233) [reply packages/vrender-components/src/indicator/indicator.ts:258]: 编译产物看了一下应该也是 ** 的写法，感觉现代的浏览器应该能正常支持？

**Review by xile611 — APPROVED** [review]

**Pull request merged**